consensus/scp/tests/mock_network/mod.rs

```diff
@@ -379,24 +302,25 @@ impl SCPNode {
 
         let thread_shared_data = Arc::clone(&node.shared_data);
         let thread_local_node = Arc::clone(&node.local_node);
+        let mut allow_propose: bool = true;
```

Comment:
What is the purpose of `allow_propose`? can you please add a comment explaining that?